packages/vite-plugin-checker/src/main.ts

```diff
@@ -73,12 +73,9 @@ export default function Plugin(userConfig: UserPluginConfig): Plugin {
         execPath: process.execPath,
       })
 
-      Promise.all(checkers.map((checker) => spawnChecker(checker, userConfig, localEnv))).then(
-        (exitCodes) => {
-          const exitCode = exitCodes.find((code) => code !== 0) || 0
-          process.exit(exitCode)
-        }
-      )
+      const exitCodes = await Promise.all(checkers.map((checker) => spawnChecker(checker, userConfig, localEnv)));
```

Comment:
if we `await` here. The checker and the rollup build will run sequentially. Is it better to run parallelly? 🤔

Reply:
Depends on what the purpose of running the checker on build is? I can see both being valid perhaps? Even if you wanted to run it in parallel, you could still utilize async/await (just spawn a new async function at the end). But yeah, I guess the current behavior should perhaps remain for now.

Reply:
Maybe we can add an option in the future to set whether run check sequentially. We can keep the parallel behaviour for now. 😉

Reply:
Sounds good. I can make the changes to this branch tomorrow, unless you want to make them yourself sooner than that, up to you.

Reply:
Sounds good. I can make the changes to this branch tomorrow, unless you want to make them yourself sooner than that, up to you.

Reply:
I'll wait for you. No hurry.

Reply:
I changed it so that it's completely async. I am not sure if rollup waits (or even can wait) on something like this to finish before the process exits (if the build is faster than the checking)? If it doesn't automatically wait, then throwing an error or killing the process is moot since you will get varying behavior depending on the speed of your build/checking 🤔

Reply:
Good catch! I just made a little test: change the tsc command `tsc --noEmit` to `sleep 20`  and the build phase wait until 20 seconds past then exit elegantly. So I guess Rollup had done something to wait for the child processes exiting first.

Reply:
Ah nice test, I'll update to still throw then!

Reply:
Though, if we throw inside a non-awaited Promise, the error just gets ignored, so I guess we have to go with `process.exit` then.

Reply:
What about use `Promise.all([...]).catch(f)`? It's non-awaited but we can catch the error.

Reply:
yeah we can do that, but then what? We could log a message I guess? I don't know what happens if you re-throw inside of the catch? 🤔

Reply:
I take a look into #60 again and find I misunderstand the initial issue context. There are 2 possible conditions.

1. checkers processes exit early than the build process
2. checkers processes exit later than the build process

For condition 1:

In #60,  no checker is enabled and the code runs into

https://github.com/fi3ework/vite-plugin-checker/blob/ba929c57e50b8c53fee7eaf1f9b9e68d732d45e1/packages/vite-plugin-checker/src/main.ts#L79

If we change this to only invoke `process.exit()` when the exit code is not `0`, the early exit in #60 should be fixed. (After I typed these I found you already have done it!)

For condition 2:

I made a little test yesterday and Rollup will wait for the child processes to exit first. So no extra things we need to do for this condition.

--- 

So in conclusion. I think you almost finish it. We only have the following points of uncertainty:

1. In `Promise.all()`, it contains the processes exit codes. I think it will not throw and we can ignore the catch code. The checker will console the error message if it does not exit with 0.
2. As condition 2 works fine. I think we do not need to `await` the `Promise.all()`. 🤔

Thanks! 🤗

Reply:
CI failed at `build` step 👀

Reply:
Sounds good, thanks for the write-up! Very odd that the build is failing though, it's not even on a line that I touched :thinking:

Reply:
oooh it was because of a missing semicolon. I can strongly recommend you to add some stricter linting to this project. Semi-colons should be mandatory on all the lines :sweat_smile:

Reply:
I'd suggest you to take a look at https://www.npmjs.com/package/eslint-config-airbnb-base. But either way, this MR should be good now.

Reply:
Yeah, this PR can be merged now. I'll do some more work and then release a version. (still haven't use semantic-release now 😅)

It's because in this line

```ts
      })

      // spawn an async runner that we don't wait for in order to avoid blocking the build from continuing in parallel
      (async () => {
```

There should be a semicolon before `(async` because it's a new line that starts with `)`. Non-semi is also acceptable (such as Vite). 

Thanks for your PR! 🤗